### PR TITLE
New v4 client & Go 1.12 for u-root

### PR DESCRIPTION
**Pull request by hugelgupf**
@insomniacslk 🎉🎉

### CODE REVIEW

File: cmds/pxeboot/pxeboot.go [hunk 1]
```diff
@@ -62,12 +65,26 @@ func Netboot(ctx context.Context, ifaceNames string) error {
...
+			}
+
+			// Cancel other DHCP requests in flight.
+			cancel()
+			log.Printf("Got configuration: %s", img)
+
+			if *dryRun {
+				img.ExecutionInfo(log.New(os.Stderr, "", log.LstdFlags))
 				return nil
+			} else if err := img.Execute(); err != nil {
```
Comment (rminnich) [inline cmds/pxeboot/pxeboot.go]: do you need this else?

File: cmds/pxeboot/pxeboot.go [hunk 2]
```diff
@@ -114,29 +131,13 @@ func Boot(lease dhclient.Lease) error {
...
+	return getBootImage(uri, lease.Link().Attrs().HardwareAddr, ip)
 }
 
 func main() {
 	flag.Parse()
 
-	ctx, cancel := context.WithTimeout(context.Background(), dhcpTries*dhcpTimeout)
-	defer cancel()
-	if err := Netboot(ctx, "eth0"); err != nil {
+	if err := Netboot("eth0"); err != nil {
```
Comment (insomniacslk) [inline cmds/pxeboot/pxeboot.go]: will the interface become a flag at a later stage?

File: pkg/less/less.go [hunk 3]
```diff
@@ -45,7 +45,7 @@ const (
 
 type Less struct {
 	// src is the source file being displayed.
-	src lineio.LineReader
+	src *lineio.LineReader
```
Comment (insomniacslk) [inline pkg/less/less.go]: does this belong to another PR?

File: pkg/strace/strace_test.go [hunk 4]
```diff
@@ -39,4 +39,7 @@ func TestBasicStrace(t *testing.T) {
 	}
 
 	go c.RunTracerFromCmd(exec.Command("ls", "/etc/hosts"))
+	for range c.Records {
```
Comment (insomniacslk) [inline pkg/strace/strace_test.go]: not sure what this is for

Comment (hugelgupf) [reply pkg/strace/strace_test.go]: RunTracerFromCmd sends stuff on this channel. It'll close it when all processes exit. This is to make sure that RunTracerFromCmd actually finishes & exits before we exit the test process.

File: pkg/less/less.go (hunk 3 above)
Comment (hugelgupf) [reply pkg/less/less.go]: No, it's fixing a new go vet warning in Go 1.12.

File: cmds/pxeboot/pxeboot.go (hunk 2 above)
Comment (hugelgupf) [reply cmds/pxeboot/pxeboot.go]: Yeah, soon enough, another PR...

Comment (hugelgupf) [reply cmds/pxeboot/pxeboot.go]: I plan to make it the same thing as in dhclient.
